pkg/asyncprofiler/asyncprofiler.go

```diff
@@ -0,0 +1,122 @@
+// Copyright 2022-2023 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package asyncprofiler
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+)
+
+// Action represents the various actions available for AsyncProfiler.
+type Action string
+
+var validActions = []string{
+	"start",
+	"resume",
+	"stop",
+	"dump",
+	"check",
+	"status",
+	"list",
+	"collect",
```

Comment:
I'm not sure about all of these, but I know `collect` is not something that the async profiler itself does, it was just a meta-thing that the `profiler.sh` did. I think the only valid actions interacting with the java-agent are `start`, `stop`, and `resume`.

Reply:
You are correct about `collect`. I'll remove it and actually let me remove all the others we are not using just yet. Most of the others are valid though. Can be checked with something like `/home/vaishali/async-profiler/build/jattach 839511 load /home/vaishali/async-profiler/build/libasyncProfiler.so true <action>`

But I think it's better to add them when we add logic around using them so that it's not confusing for others. For example, checking `status` for debugging/some error handling, using `dump` to dump the output etc.